### PR TITLE
Use new Scale transformation constructor (introduced in compas 0.16)

### PR DESCRIPTION
**Pull request by tetov**
### What type of change is this?

- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [x] Upstream version bump

### Checklist

- ~~I added a line to the `CHANGELOG.rst` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).~~
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- ~~I added new functions/classes and made them available on a second-level import, e.g. `compas_fab.robots.Configuration`.~~
- ~~I have added tests that prove my fix is effective or that my feature works.~~
- ~~I have added necessary documentation (if appropriate).~~

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 requirements = [
     # Until COMPAS reaches 1.0, we pin major.minor and allow patch version updates
-    'compas>=0.11,<0.14',
+    'compas>=0.16,<0.17',
```
Comment (gonzalocasas) [inline setup.py:14]: It seems you have a very old base to work on.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -29,12 +29,15 @@ Unreleased
...
+* In :class:`compas_fab.robots.constraints.JointConstraint`, added ``tolerance_above`` and
   ``tolerance_below`` for allowing asymmetrical constraints.
-* In :class:``compas_fab.robots.Robot``, changed the ``constraints_from_configuration``
+* In :class:`compas_fab.robots.Robot`, changed the ``constraints_from_configuration``
   function with ``tolerances_above`` and ``tolerances_below``.
 * :meth:`compas_fab.robots.CollisionMesh.scale` now takes a scale factor
   instead of a :class:`compas.geometry.Scale` instance as an argument.
+* :meth:`compas_fab.robots.CollisionMesh.scale` now uses constructor
+  :meth:`compas.geometry.Scale.from_factors` to create transformation. This is to
+  adapt to a breaking change introduced in compas 0.16.
```
Comment (gonzalocasas) [inline CHANGELOG.rst:40]:
```suggestion
```

Don't add this to the changelog, it's not a change that affects the user.

**Comment by gonzalocasas** [comment]
(sorry, I notice now this is on `Draft`)

File: setup.py (hunk 1 above)
Comment (tetov) [reply setup.py:14]: Yeah I was surprised too. I’ll check what’s up.

**Comment by tetov** [comment]
Hold off on this one a bit please, I'll work through errors raised in doctests.

**Comment by tetov** [comment]
`compas_fab.robots.robot.Robot.plan_motion` doctest fails due to bug in `compas.geometry.Sphere` on 0.16.0 [travis log](https://travis-ci.com/github/tetov/compas_fab/jobs/345776924#L520)

Fixed in [this PR](https://github.com/compas-dev/compas/pull/547) but we need new release or +DOCTEST_SKIP flag.
https://github.com/compas-dev/compas/pull/547

**Comment by gonzalocasas** [comment]
> `compas_fab.robots.robot.Robot.plan_motion` doctest fails due to bug in `compas.geometry.Sphere` on 0.16.0 [travis log](https://travis-ci.com/github/tetov/compas_fab/jobs/345776924#L520)
> 
> Fixed in [this PR](https://github.com/compas-dev/compas/pull/547) but we need new release or +DOCTEST_SKIP flag.

you're correct... :( We're preparing a new patch release for 0.16.x: https://github.com/compas-dev/compas/tree/hotfix/v0.16.x

**Comment by tetov** [comment]
> you're correct... :( We're preparing a new patch release for 0.16.x: https://github.com/compas-dev/compas/tree/hotfix/v0.16.x

It happens! The error in sphere transformation is the only remaining fail in Travis so this could either sit and wait for the hotfix or we could merge it to a `compas_0.16_compat` branch to collect compatibility changes.

File: setup.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 requirements = [
     # Until COMPAS reaches 1.0, we pin major.minor and allow patch version updates
-    'compas>=0.15.6,<0.16',
+    'compas>=0.16,<0.17',
```
Comment (gonzalocasas) [inline setup.py:14]:
```suggestion
    'compas>=0.16.1,<0.17',
```

compas 0.16.1 has just been released

**Pull request merged**